src/desfire/devices/pn532.py

```diff
@@ -40,13 +44,28 @@ def __init__(self, port: str, **kwargs):
         Keyword Args:
             All keyword arguments are passed to the pyserial.Serial constructor.
         """
-
         if not _has_serial:
             raise ImportError("pyserial is required for using PN532UARTDevice")
 
         self._uart = serial.Serial(port, **kwargs)
+        self._wakeup()
+        ic, ver, rev, support = self.firmware_version()
+        logger.info("PN532 initialized, found version {ver}.{rev}.")
```

Comment:
This is not an f-string, so values for `ver` and `ref` are never replaced, it's always printed literally. For example:

´´´
2025-07-06T11:17:27.065252Z [info     ] PN532 initialized, found version {ver}.{rev}. [desfire.devices.pn532]
```